waspc/e2e-test/GoldenTest.hs

```diff
@@ -79,27 +80,52 @@ runGoldenTest goldenTest = do
           goldenOutputAbsFp
           currentOutputAbsFp
           (return ()) -- A no-op command that normally generates the file under test, but we did that in bulk above.
-        | currentOutputAbsFp <- manifestAbsFp : currentOutputAbsFps,
+        | currentOutputAbsFp <- manifestAbsFp : filesForGoldenTestAbsFps,
           let goldenOutputAbsFp = remapCurrentPathToGolden currentOutputAbsFp
       ]
   where
-    listRelevantTestOutputFiles :: FilePath -> IO [FilePath]
-    listRelevantTestOutputFiles dirToFilterAbsFp =
-      getDirFiltered (return <$> isTestOutputFileTestable) dirToFilterAbsFp >>= filterM doesFileExist
-
-    isTestOutputFileTestable :: FilePath -> Bool
-    isTestOutputFileTestable fp =
+    manifestFileName :: String
+    manifestFileName = "files.manifest"
+
+    -- Lists files that should be included in the `files.manifest`.
+    listFilesForManifest :: FilePath -> IO [FilePath]
+    listFilesForManifest dirToFilterAbsFp =
+      getDirFiltered (return <$> shouldIncludeInManifest) dirToFilterAbsFp >>= filterM doesFileExist
+
+    -- Lists files that should be compared in golden tests.
+    listFilesForGoldenTest :: FilePath -> IO [FilePath]
+    listFilesForGoldenTest dirToFilterAbsFp =
+      getDirFiltered (return <$> shouldIncludeInGoldenTest) dirToFilterAbsFp >>= filterM doesFileExist
+
+    -- Determines if a file should be included in the `files.manifest`.
+    shouldIncludeInManifest :: FilePath -> Bool
+    shouldIncludeInManifest fp =
       takeFileName fp
-        `notElem` [ ".waspinfo",
-                    "node_modules",
+        `notElem` [ ".DS_Store",
                     "dev.db",
                     "dev.db-journal",
-                    "package-lock.json",
                     ".gitignore",
-                    ".DS_Store",
+                    ".waspinfo",
+                    "package-lock.json",
+                    "node_modules",
                     "tsconfig.tsbuildinfo"
                   ]
 
+    -- Determines if a file should be included in golden test comparisons.
+    shouldIncludeInGoldenTest :: FilePath -> Bool
+    shouldIncludeInGoldenTest fp =
```

Comment:
These names come from the implementation (solution domain), not the intent (problem domain). I'm sure there's a section in clean code that talks about this, but I'm not sure where :)

Anyway, I think the names should reflect what we want to do: check for existance and compare contents. So something like:
```
shouldCheckFileContents fp = ...
shouldCheckIfFileExists fp = ...
``` 
Or something along those lines.